includes/class-wc-gateway-amazon-payments-advanced.php

```diff
@@ -2815,13 +2827,32 @@ public function load_scripts_globally_if_button_enabled_on_cart( $load_scripts )
 	}
 
 	/**
-	 * Loads scripts on product pages when the setting to display Amazon Pay button on products is enabled.
+	 * Loads scripts on product pages when the setting to display Amazon Pay button on products is enabled
+	 * and the product is purchasable and not out of stock.
 	 *
 	 * @param bool $load_scripts Whether to load Amazon Pay scripts or not.
 	 * @return bool
 	 */
 	public function load_scripts_on_product_pages( $load_scripts ) {
-		return $load_scripts ? $load_scripts : is_product() && $this->is_product_button_enabled();
+		if ( ! is_product() ) {
+			return $load_scripts;
+		}
+
+		if ( ! $this->is_product_button_enabled() ) {
+			return $load_scripts;
+		}
+
+		$product = wc_get_product();
```

Comment:
Those 2 ifs are structurally code duplicates of what's we are changing in 'maybe_separator_and_checkout_button_single_product'.  Can we refactor that?

Reply:
Yea makes sense. Refactored!